Cython/Compiler/Optimize.py

```diff
@@ -3499,15 +3499,22 @@ def _inject_unicode_predicate(self, node, function, args, is_unbound_method):
             return node
         uchar = ustring.arg
         method_name = function.attribute
-        if method_name in ('istitle', 'isprintable'):
+        if method_name == 'istitle':
             # istitle() doesn't directly map to Py_UNICODE_ISTITLE()
-            # isprintable() is lacking C-API support in PyPy
             utility_code = UtilityCode.load_cached(
                 "py_unicode_%s" % method_name, "StringTools.c")
             function_name = '__Pyx_Py_UNICODE_%s' % method_name.upper()
         else:
-            utility_code = None
-            function_name = 'Py_UNICODE_%s' % method_name.upper()
+            # None of these are defined in the Limited API
+            utility_code = TempitaUtilityCode.load_cached(
+                "py_unicode_predicate", "StringTools.c",
+                context=dict(
+                    # isprintable() is lacking C-API support in PyPy
+                    generate_for_pypy=method_name == "isprintable",
+                    method_name=method_name
+                )
```

Comment:
I think we can also do this entirely in the template code. No need to handle PyPy specially here.
```suggestion
                context=dict(method_name=method_name)
```

(In fact, we could probably also include the `istitle()` case in the same way.)